### DISCUSSION TITLE
Thalamus brain suicide by lack of water

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14498] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

On some wrecks, broken/open ducts let water flow drain to rooms below and the thalamus dies to a lack of water

There's honestly no reason for thalamus *structures* (not leucocytes) to be taking damage when not in water because it just makes organs die at random. 

Kastrull
![image](https://github.com/user-attachments/assets/0f087cb6-9979-4e13-b36b-f2ee1ea53df3)

### Reproduction steps

.

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Why not flood wrecked hulls on spawn?
They also take 100 years to fill and lag all that time

**Reply by ubertpendragon** [reply]
Rooms are flooded at random when the wreck spawns, some room may be wet some may be dry, in dry room thalamus organs will just die

**Comment by BlackMoons** [comment]
Simple solution might be just make the brain leak ~brain fluid~ water into the hull its in.

**Reply by SomeRandomNoobKekeke** [reply]
And lag even more 💀

**Reply by ubertpendragon** [reply]
> Simple solution might be just make the brain leak brain fluid water into the hull its in.

That would break a bunch of custom builds I know

**Comment by Regalis11** [comment]
> There's honestly no reason for thalamus structures (not leucocytes) to be taking damage when not in water because it just makes organs die at random.

The reason is to give the players an alternative way to destroy the organs (by repairing the walls and draining the hulls)

**Reply by ubertpendragon** [reply]
This would be possible if the wrecks were actually designed to be repairable. 

Most of them have no hulls in the ballasts and thus can't be drained

Not to mention the amount of effort that it takes to fix a wreck *with thalamus alive*, compared to just killing the brain

**Reply by Regalis11** [reply]
Portable pumps exist

**Reply by ubertpendragon** [reply]
So do guns and they are more easily available

A thalamus brain that can't be killed by conventional means would be interesting as a special mission but no one ever drained a thalamus to death as a means to dealing with wrecks

I can't think a way of making thalamus AI work in custom maps either in the rare case you want to do that

**Comment by itchyOwl** [comment]
The brain shouldn't spawn in rooms that can dry out (without breaching the walls or opening the doors/hatches), unless there's no other rooms to be found, in which case there should be a warning.

I think it's just natural that they need water to survive. But the technical reason is that if the room is dry, the cells protecting the brain will die. So dry brain rooms are an issue even if we'd make the brain not take damage from the water.

I'll try to see if there's a bug here, and if there's something more that can be done to prevent the brain from spawning into rooms like that.

**Comment by itchyOwl** [comment]
> Why not flood wrecked hulls on spawn?

They are randomly flooded and the brain rooms are always flooded, but that doesn't help if there's an open gap in the floor.

**Comment by itchyOwl** [comment]
Seems that the check was bugged: the gaps aren't yet open when we crete the wreck. The gaps appear only after that. But checking the item condition works. Fixed in FakeFishGames/Barotrauma-development#5767.

**Reply by ubertpendragon** [reply]
This won't really solve the issue as we experienced when making wrecks:

It's too easy to leave a duct wired to a water detector, either by habit or just by using a wreck converted. I'm convinced many wrecks have this exact problem without the author knowing

Making the brain not die to lack of water feels like a safer solution; even if the Leucocytes die you'll avoid all the cases of thalamus suicide. The air vulnerability is NEVER in play so in practice it does not exist

**Reply by itchyOwl** [reply]
I also made the Thalamus jam the doors now, but that's not enough either, if there's no available brain rooms in the first place, which seems to be the case with some of the new wrecks.

Making the brain not take damage in air would indeed be a bullet-proof solution to it not dying prematurely, but it wouldn't address wrong room being selected in the first place. Seems that there's multiple issues related to this with the new wrecks that needs to be solved. And we also need to tell the sub builders when there's no valid brain rooms (in which case a random room is chosen).

I tested that draining the brain room (without any portable pumps) is quite feasible, but only in certain wrecks, like Kastrull. And only when the critical hulls don't flood. I think I'll want to try improving this intended feature so that it would not be "next to impossible" to do.

However, I think it's not that important that the brain actually dies when the room is drained. It's more important that the only way to get around the leucocytes is not to kill them. So it's worth considering to make the brain not take damage in dry hulls. But I'll have to do more testing to be able to decide on this.

**Comment by itchyOwl** [comment]
Should be fully addressed in https://github.com/FakeFishGames/Barotrauma-development/pull/5767. Closing
